SmartDeviceLink/private/SDLStreamingVideoLifecycleManager.m

```diff
@@ -659,6 +687,216 @@ - (void)sdl_hmiStatusDidChange:(SDLRPCNotificationNotification *)notification {
+        [notificationCenter addObserver:self selector:@selector(sdl_hmiStatusDidChange:) name:SDLDidChangeHMIStatusNotification object:nil];
+        [notificationCenter addObserver:self selector:@selector(sdl_appStateDidUpdate:) name:UIApplicationDidBecomeActiveNotification object:nil];
+        [notificationCenter addObserver:self selector:@selector(sdl_appStateDidUpdate:) name:UIApplicationWillResignActiveNotification object:nil];
+    }
+}
+
+- (void)sdl_unsubscribeNotifications {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+    self.isSubscribedForNotifications = NO;
+}
+
+#pragma mark - Subscribed notifications
+
+- (void)sdl_videoStreamingCapabilityDidUpdate:(SDLSystemCapability *)systemCapability {
+    SDLVideoStreamingCapability *videoCapability = systemCapability.videoStreamingCapability;
+
+    if (!videoCapability) {
+        SDLLogD(@"Empty video capabilities notification received (%@)", self.videoStreamStateMachine.currentState);
+    } else {
+        SDLLogD(@"Video capabilities notification received (%@): %@", self.videoStreamStateMachine.currentState, videoCapability);
+    }
+
+    if ([self.videoStreamStateMachine.currentState isEqualToEnum:SDLVideoStreamManagerStateStarting]) {
+        [self sdl_applyVideoCapabilityWhileStarting:videoCapability];
+    }
+    else if ([self.videoStreamStateMachine.currentState isEqualToEnum:SDLVideoStreamManagerStateReady]) {
+        [self sdl_applyVideoCapabilityWhenStreaming:videoCapability];
+    }
+}
+
+#pragma mark - Video capability logic
+
+- (void)sdl_applyVideoCapabilityWhileStarting:(nullable SDLVideoStreamingCapability *)videoCapabilityUpdated {
+    SDLVideoStreamingCapability *videoCapability = (nil == videoCapabilityUpdated) ? [self sdl_defaultVideoCapability] : videoCapabilityUpdated;
+    NSArray<SDLVideoStreamingCapability *> *capabilityMatches = [self matchVideoCapability:videoCapability];
+
+    if (capabilityMatches.count == 0) {
+        // use default video capability if no match found
+        capabilityMatches = @[[self sdl_defaultVideoCapability]];
+    }
+
+    SDLVideoStreamingCapability *matchedVideoCapability = capabilityMatches.firstObject;
+    if (capabilityMatches.count > 1) {
+        // add the remaining capabilities that match video streaming criteria
+        matchedVideoCapability.additionalVideoStreamingCapabilities = [capabilityMatches subarrayWithRange:NSMakeRange(1, capabilityMatches.count - 1)];
+    } else {
+        // add any other possible capabilities here if needed
+        matchedVideoCapability.additionalVideoStreamingCapabilities = @[];
+    }
+
+    // create and send SDLOnAppCapabilityUpdated notification
+    SDLAppCapability *appCapability = [[SDLAppCapability alloc] initWithAppCapabilityType:SDLAppCapabilityTypeVideoStreaming videoStreamingCapability:matchedVideoCapability];
```

Comment:
The proposal was not very clear on what properties should get set on `OnAppCapabilityUpdated.appCapability. videoStreamingCapability` so we have decided the mobile libraries should not set the root `videoStreamingCapability` but only the `additionalVideoStreamingCapabilities` as outlined below:

1. Mobile checks both the root `videoStreamingCapability` and the `additionalVideoStreamingCapabilities` to see which ones it can support. 
2. Mobile sends an `OnAppCapabilityUpdated` with only the `VideoStreamingCapability.additionalVideoStreamingCapabilities` parameter set. The  `additionalVideoStreamingCapabilities` should be set with all the capabilities it can support as determined by step 1.